server/routes/signup/signupHandler.ts

```diff
@@ -0,0 +1,96 @@
+import { NextFunction, Request, Response } from "express";
+import * as EmailValidator from "email-validator";
+
+import DBHandler from "./signupDBHandler";
+import { hash } from "bcrypt";
+import { ParentSignup } from "./signupTypes";
+
+export default class Handler {
+  private dbHandler: DBHandler;
+
+  constructor() {
+    this.dbHandler = new DBHandler();
+  }
+
+  getDBHandler() {
+    return this.dbHandler;
+  }
+
+  signupValidation = (req: Request, res: Response, next: NextFunction) => {
+    const minPasswordLength = 4;
+    const { comments, ...inputs } = req.body;
+
+    const hasNullInput = Object.values(inputs).some(
+      (value) => value === null || value === "" || value === 0
+    );
+
+    if (
+      hasNullInput ||
+      !EmailValidator.validate(req.body.email) ||
+      req.body.password.length < minPasswordLength
+    ) {
+      res
+        .status(401)
+        .json({ message: "Please fill out all the required fields" });
+    } else {
+      next();
+    }
+  };
+
+  signupParent = async (req: Request, res: Response) => {
+    try {
+      const existingParent = await this.dbHandler.existingParent(
+        req.body.email
+      );
+
+      if (existingParent) {
+        return res.status(401).json({ message: "Email is already in use" });
+      }
+    } catch (error) {
+      console.log(error);
+      res.status(401).json({ message: "Failed sign up" });
```

Comment:
change error code, existing parents -> 401, server error -> 403/500 or any other 5xx you want